### ISSUE TITLE
[Bug]: ilamb unable to submit due to dependency check failure attributable to inconsistent resolving of ts_num_years 

### ISSUE THREAD

**Issue by wlin7** [E3SM-Project/zppy#599]
### What happened?

On running ilamb for v3.LR.historical_0101, with the following ilamb section in the cfg file
```
[ilamb]
active = True
nodes = 8
walltime = "2:00:00"
partition = compute
short_name = 'v3.LR.historical_0101'
ts_num_years = 5
years = "1985:2014:30"

```
zppy would prompt

```
ilamb_1985-2014
...skipping because of dependency status file missing
   /lcrc/group/e3sm2/ac.wlin/E3SMv3/v3.LR.historical_0101/post/scripts/ts_land_monthly_1985-2014-0005.status
```
`ts` files for `land_monthly`  was set to be every 5 years, which were generated properly. `ts_num_years = 5` was set accordingly for the ilamb section. The dependency is expected to be on every 5 years from 1985 to 2014. But the above apparently is checking on `1985-2014-0005`, in which the number of years `0005` does not match the length from `start year (1985) ` to `end year (2014).`

### What machine were you running on?

chrysalis

### Environment

zppy in e3sm_unified_1.10

### What command did you run?

```bash
zppy -c post.v3.LR.historical_0101.cfg.new
```

### Copy your cfg file

```bash
/home/ac.wlin/E3SM/utils/post.v3.LR.historical_0101.cfg.new

The ilamb section of the cfg file has been changed to use `ts_num_years = 30`, with `ts_num_years = 5` commented out to allow ilamb to proceed. If keeping `ts_num_years = 5`, it would produce the message that ilamb is skipped.
```

### What jobs are failing?

```bash
ilamb
```

### What stack trace are you encountering?

```bash
N/A. Unable to submit due to inconsistent dependency check.
```

**Comment by forsyth2** [comment]
Thanks @wlin7. This points to a larger problem of running any task on a time period consisting of _many_  time periods for its dependent task.

In https://github.com/E3SM-Project/zppy/blob/main/zppy/ilamb.py:
```
            dependencies.append(
                os.path.join(
                    scriptDir,
                    "ts_%s_%04d-%04d-%04d.status"
                    % (
                        c["ts_land_subsection"],
                        c["year1"],
                        c["year2"],
                        c["ts_num_years"],
                    ),
                ),
            )
```
Similarly in https://github.com/E3SM-Project/zppy/blob/main/zppy/e3sm_diags.py:
```
                        dependencies.append(
                            os.path.join(
                                scriptDir,
                                "ts_%s_%04d-%04d-%04d.status"
                                % (ts_sub, start_yr, end_yr, c["ts_num_years"]),
                            )
                        )
```
We'd probably need to replace each block of that type with a for-loop of dependency addition incrementing the start_yr and end_year by ts_num_years each time.

**Comment by forsyth2** [comment]
@wlin7 Did setting `ts_num_years = 30` work for you? It seems to me that wouldn't work either because you never ran a 30 year time-series for the `ts` task.

**Comment by forsyth2** [comment]
Resolved by #603
